source/common/network/address_impl.cc

```diff
@@ -351,9 +357,10 @@ PipeInstance::PipeInstance(const sockaddr_un* address, socklen_t ss_len)
   } else {
     friendly_name_ = address->sun_path;
   }
+  this->mode = mode;
```

Comment:
can we check if `abstract_namespace_` is true here and throw if the mode is set there?